### PR TITLE
Add verify.py for iosxe eigrp folder containing 02 APIs for EIGRPv4 and EIGRPv6 

### PR DESCRIPTION
**Pull request by cherifimehdi**
Hello,

I noticed there is no verify.py file and verify apis for ios/iosxe eigrp protocol

I added verify.py file at pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp containing 02 APIs for EIGRPv4 and EIGRPv6:

        * verify_eigrp_interfaces
	    * verify_eigrp_neighbors

I added all the tests at pkgs/sdk-pkg/src/genie/libs/sdk/apis/tests/iosxe/eigrp/verify containing the test results for the 02 APIs in verify.py

I added the changelog_add_eigrp_verify.py_apis_202405031904 file at pkgs/sdk-pkg/changelog folder

Thank you for reviewing my contribution

### CODE REVIEW

**Comment by cherifimehdi** [comment]
Hello @ThomasJRyan @Lelor can you please tell me why there is a problem in test step?  there is Some checks were not successful but for me it is OK since I added previously get.py for EIGRP without any problem. Thank you

**Comment by ThomasJRyan** [comment]
image

Looks to be caused by your branch being out of date. We're currently on version 24.2. Pull the latest `master` branch into yours and all should be good in the world

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:29]: I think we'd want to return `False` here?
```suggestion
               log.error(f"Sorry,no data for the provided 'AS = {AS_N}' and/or 'vrf = {vrf}' !")
               return False
```

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:32]: Actually, at this point I think we can just return no? No need to set a variable
```suggestion
            return set(interfaces_list).issubset(interfaces)
```

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:33]: I think this would fit better under your `SchemaEmptyParserError`

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:36]:
```suggestion
         log.error(f"Please, provide a valid format for AS, interfaces_list, ip and/or vrf")
```

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:46]: This isn't in the correct order, and you're missing `max_time` and `check_interval`

Comment (ThomasJRyan) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:46]: Also, it should be `neighbors`

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:52]: Same thing about being `assert`s and outside the loop

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:68]:
```suggestion
         log.error(f"Please, provide a valid format for AS, Neighbors, ip and/or vrf")
```

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:53]: Same thing about moving this out the loop

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:57]: `continue` or return `False`

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:62]: Since you're returning in the previous line, you don't need this `else`

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:61]:
```suggestion
               return False
```

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:64]:
```suggestion
            return set(Neighbors).issubset(eigrp_neighbors)
```

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:63]: Any reason to not just do
```suggestion
               eigrp_neighbors = response.q.get_values("eigrp_nbr")
```
?

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:31]: Any reason to not just do
```suggestion
               interfaces = response.q.get_values('interface')
```
?

**Review by ThomasJRyan — CHANGES_REQUESTED** [review]
Thank you for the contribution. Unfortunately I think this PR will need some work before we can merge it in

**Comment by cherifimehdi** [comment]
Hi @ThomasJRyan, thanks for reviewing my PR. I went through all the changes and tested it, it works perfectly.
Thanks for reviewing my post.

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:1]: Hi @ThomasJRyan I fixed all the issues. Thank you

**Comment by cherifimehdi** [comment]
Hello @ThomasJRyan @Lelor @Sohan are the changes I made ok. Because I addressed all them. Thank you

**Comment by cherifimehdi** [comment]
Hi @ThomasJRyan are the changes I made ok based on your review? Thanks

**Comment by cherifimehdi** [comment]
Hello @ThomasJRyan Thank you for your valuable comments. Now I used Black instead of Pylint and normally it is reformatted. I have made all the necessary changes based on your suggestions and feedback. Thanks for reviewing my PR

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:68]: These timeouts should be removed

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:127]: Same with these ones

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:41]: This could also be an assertion

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (ThomasJRyan) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py:102]: Same

**Review by ThomasJRyan — APPROVED** [review]
Small things, but nothing worth blocking the PR further over. Thank you for the contribution!

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py]: I removed the timeout.sleep()

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py]: I removed the timeout.sleep()

**Comment by cherifimehdi** [comment]
Hi @ThomasJRyan Thank you very much for your valuable remarks and taking time to review my PR and accepting my contribution.
By the way I addressed the suggested changes on timeout.sleep().
Thank you sir

**Comment by ThomasJRyan** [comment]
No problem! I'll get @Lelor on this review as soon as possible so we can get you all merged in

**Review by Lelor — APPROVED** [review]

**Pull request merged**
